### PR TITLE
Devop 1845 compose evm simplified improvemets

### PR DESCRIPTION
**Pull request by otoumas**
- Updated CHANGELOG
- Updated restart policy on all services
- Migration from 1.2.* to 1.3.0 project structure
- Updated .env.template files default values
- Added upgrade.sh script that will:
    - Create a bk of original .env file
    - Add new env vars to .env file
    - Auto update some env vars
    - Check optional vars for differences and propose update to the user

### CODE REVIEW

File: scripts/upgrade.sh [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+conditional_update_vars=(
+  "ZEN_OPTS"
+  "SCNODE_FORGER_MAXCONNECTIONS"
+  "SCNODE_NET_MAX_IN_CONNECTIONS"
+  "SCNODE_NET_MAX_OUT_CONNECTIONS"
+  "SCNODE_LOG_FILE_LEVEL"
+)
+
+# Read the .env.template file line by line, skip blanc lines and comments, store each of the other lines in an array
```
Comment (ysibirski) [inline scripts/upgrade.sh]: blank typo in comment :)

File: scripts/upgrade.sh [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  [ "${line:0:1}" = "#" ] && continue
+  env_template_lines+=("$line")
+done <"$ENV_FILE_TEMPLATE"
+
+# Append new env vars to .env file
+echo -e "\n\033[1m=== Appending new env vars to .env file ===\033[0m"
+for line in "${env_template_lines[@]}"; do
+  var_name=$(echo "$line" | cut -d'=' -f1)
+  if ! grep -q "^$var_name=" "$ENV_FILE"; then
+    echo "$line" >>"$ENV_FILE"
```
Comment (ysibirski) [inline scripts/upgrade.sh]: Be careful with this. IF your ENV_FILE is not ending with new line char your first ECHO will append `var_name=` to the same last line in the file. Workaround if needed can be `echo -e "\n${line}" >> ....`

File: scripts/upgrade.sh [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  "SCNODE_LOG_FILE_LEVEL"
+)
+
+# Read the .env.template file line by line, skip blanc lines and comments, store each of the other lines in an array
+echo -e "\n\033[1m=== Reading .env.template file ===\033[0m"
+while IFS= read -r line; do
+  [ -z "$line" ] && continue
+  [ "${line:0:1}" = "#" ] && continue
+  env_template_lines+=("$line")
+done <"$ENV_FILE_TEMPLATE"
```
Comment (ysibirski) [inline scripts/upgrade.sh]: I am a fan of {} around vars but it is up to you :). If you decided to fix it there are more occurrences of this

File: scripts/upgrade.sh [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+done
+
+# Update the values of the auto update variables
+echo -e "\n\033[1m=== Updating the values of the auto update variables ===\033[0m"
+for line in "${env_template_lines[@]}"; do
+  var_name=$(echo "$line" | cut -d'=' -f1)
+  for item in "${auto_update_vars[@]}"; do
+    if [[ "$item" == "$var_name" ]]; then
+      sed -i "/^$var_name=/c\\$line" "$ENV_FILE"
+      break
```
Comment (ysibirski) [inline scripts/upgrade.sh:77]: Will this exit from inner loop and move onto the next line in outer loop?

File: scripts/upgrade.sh [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+# Update the values of the conditional update variables if approved by the user
+echo -e "\n\033[1m=== Updating the values of the conditional update variables ===\033[0m"
+for line in "${env_template_lines[@]}"; do
+  var_name=$(echo "$line" | cut -d'=' -f1)
+  for item in "${conditional_update_vars[@]}"; do
+    if [[ "$item" == "$var_name" ]]; then
+      if ! grep -q "^$line" "$ENV_FILE"; then
+        echo -e "\nThe value of $var_name in the .env file is different from the value in the .env.template file."
+        echo -e "Current value: \033[1m$(grep "^$var_name=" "$ENV_FILE")\033[0m"
+        echo -e "New value: \033[1m$line\033[0m\n"
```
Comment (ysibirski) [inline scripts/upgrade.sh]: You may wan to add {} here in case of whitespaces and special chars in the value of a $line

File: scripts/upgrade.sh [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+      break
+    fi
+  done
+done
+
+echo ".env upgrade completed successfully"
+
+echo "Please review the changes in the .env file, if there is anything wrong you can restore the backup file .env.bk"
+
+echo "Run cd deployments/${role_value}/${network_value} && docker-compose down && docker-compose up -d to apply the changes."
```
Comment (ysibirski) [inline scripts/upgrade.sh]: You can use single ECHO here if you want by utilizing HEREDOC

**Review by ysibirski — APPROVED** [review]
Some non blocking comments

File: scripts/upgrade.sh (hunk 4 above)
Comment (otoumas) [reply scripts/upgrade.sh:77]: just the inner loop

File: docs/FORGER.md [hunk 7]
```diff
@@ -65,10 +65,15 @@ This directory will be mounted into the zend container and used to seed the node
```
Comment (lander86) [inline docs/FORGER.md:65]: `docker compose exec zend gosu user zen-cli getblockcount` you need to be in the right directory in order to execute " docker compose exec" command and now you've removed the "cd" statement. Let's add the "-f" parameter also here, like `docker compose -f deployments/forger/[eon|gobi]/docker-compose.yml exec zend gosu user zen-cli getblockcount`

File: env/.env.rpc.eon.template [hunk 8]
```diff
@@ -57,5 +57,5 @@ SCNODE_WALLET_GENESIS_SECRETS=
 SCNODE_WALLET_SEED=
 SCNODE_WALLET_MAXTX_FEE=10000000
```
Comment (lander86) [inline env/.env.rpc.eon.template:60]: Let's update also here the SCNODE_NET_MAX_IN_CONNECTIONS, SCNODE_NET_MAX_OUT_CONNECTIONS and add SCNODE_FORGER_MAXCONNECTIONS=100. Double check the compose is ok about this.

File: env/.env.rpc.gobi.template [hunk 9]
```diff
@@ -56,5 +56,5 @@ SCNODE_WALLET_GENESIS_SECRETS=
 SCNODE_WALLET_SEED=
 SCNODE_WALLET_MAXTX_FEE=10000000
```
Comment (lander86) [inline env/.env.rpc.gobi.template:59]: Let's update also here the SCNODE_NET_MAX_IN_CONNECTIONS, SCNODE_NET_MAX_OUT_CONNECTIONS and add SCNODE_FORGER_MAXCONNECTIONS=100. Double check the compose is ok about this.

**Review by lander86 — CHANGES_REQUESTED** [review]
Minor comments

**Review by ysibirski — APPROVED** [review]
LGTM

**Review by lander86 — APPROVED** [review]
LGTM

**Pull request merged**